src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/Namespace/SetAzureEventHubNamespace.cs

```diff
@@ -22,65 +22,141 @@ namespace Microsoft.Azure.Commands.EventHub.Commands.Namespace
+            HelpMessage = "EventHub Namespace Location.",
+          ParameterSetName = AutoInflateParameterSet)]
         [ValidateNotNullOrEmpty]
         public string Location { get; set; }
 
         [Parameter(
-          Position = 3,
           Mandatory = false,
+          Position = 3,          
+          ValueFromPipelineByPropertyName = true,
+          HelpMessage = "Namespace Sku Name.",
+          ParameterSetName = NamespaceParameterSet)]
+        [Parameter(
+          Mandatory = false,
+          Position = 3,
           ValueFromPipelineByPropertyName = true,
-          HelpMessage = "Namespace Sku Name.")]
-                [ValidateSet(SKU.Basic,
+          HelpMessage = "Namespace Sku Name.",
+          ParameterSetName = AutoInflateParameterSet)]
+        [ValidateSet(SKU.Basic,
           SKU.Standard,
           SKU.Premium,
           IgnoreCase = true)]
         public string SkuName { get; set; }
 
         [Parameter(          
           Mandatory = false,
+            Position = 4,
           ValueFromPipelineByPropertyName = true,
-          HelpMessage = "The eventhub throughput units.")]
+          HelpMessage = "The eventhub throughput units.",
+          ParameterSetName = NamespaceParameterSet)]
+        [Parameter(
+          Mandatory = false,
+            Position = 4,
+          ValueFromPipelineByPropertyName = true,
+          HelpMessage = "The eventhub throughput units.",
+          ParameterSetName = AutoInflateParameterSet)]
         public int? SkuCapacity { get; set; }
 
         [Parameter(Mandatory = false,
+            Position = 5,
             ValueFromPipelineByPropertyName = true,            
-            HelpMessage = "Disable/Enable Namespace.")]
-        public Management.EventHub.Models.NamespaceState? State { get; set; }
+            HelpMessage = "Disable/Enable Namespace.",
+          ParameterSetName = NamespaceParameterSet)]
+        [Parameter(Mandatory = false,
+            Position = 5,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Disable/Enable Namespace.",
+          ParameterSetName = AutoInflateParameterSet)]
+        public Models.NamespaceState? State { get; set; }
 
         [Parameter(Mandatory = false,
+            Position = 6,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Hashtables which represents resource Tag.",
+          ParameterSetName = NamespaceParameterSet)]
+        [Parameter(Mandatory = false,
+            Position = 6,
             ValueFromPipelineByPropertyName = true,
-            HelpMessage = "Hashtables which represents resource Tag.")]
+            HelpMessage = "Hashtables which represents resource Tag.",
+          ParameterSetName = AutoInflateParameterSet)]
         public Hashtable Tag { get; set; }
 
+        /// <summary>
+        /// Indicates whether AutoInflate is enabled.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
```

Comment:
@v-Ajnava this parameter should be mandatory in `AutoInflateParameterSet`, should not have a position, nor should it accept a value from pipeline